crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -20,18 +20,39 @@ module Api
   class Upgrade < Tableless
     def status
       {
-        crowbar: crowbar_upgrade_status,
-        checks: check
-      }
+        crowbar: crowbar_upgrade_status
+      }.merge!(check)
     end
 
     def check
       {
-        sanity_checks: sanity_checks,
-        maintenance_updates_installed: maintenance_updates_installed?,
-        clusters_healthy: clusters_healthy?,
-        compute_resources_available: compute_resources_available?,
-        ceph_healthy: ceph_healthy?
+        checks: {
+          sanity_checks: {
+            required: true,
+            passed: sanity_checks.empty?,
+            errors: sanity_check_errors
+          },
+          maintenance_updates_installed: {
+            required: true,
+            passed: maintenance_updates_status[:passed],
```

Comment:
Unless we also wanted to be able to return warnings from these checks, in which case the whole structure needs a redesign, because a single value for `required` would not be applicable to all warnings / errors.  For example:

```
{
  "checks": {
    ....
    "maintenance_updates_installed": {
      "warnings": [
        {
          "code": "maintenance_updates_metadata_old",
          "data": [
            "Updates repo metadata is 3 days old."
          ],
          "help": "Make sure maintenance update repo metadata is refreshed via zypper refresh, or turn on auto-refresh via zypper mr -r."
        }
      ],
      "errors": [
        {
          "code": "maintenance_updates_installed",
          "data": [
            "ZYPPER_EXIT_INF_UPDATE_NEEDED: patches available for installation."
          ],
          "help": "make sure maintenance updates are installed"
        }
      ]
    },
    ....
  }
}
```